packages/resolver/lib/sources/truffle/index.ts

```diff
@@ -68,6 +73,30 @@ export class Truffle implements ResolverSource {
         this.options.compilers
       );
       return { body: addressSource, filePath: importPath };
+    } else if (importPath === `truffle${path.sep}Console.sol`) {
+      // calculating this in webpack env breaks
+      let unbundledGanacheConsoleSol;
+      // @ts-ignore
+      if (typeof BUNDLE_VERSION === "undefined") {
+        unbundledGanacheConsoleSol = path.resolve(
+          path.join(
+            require.resolve("@ganache/console.log"),
+            "..",
+            "..",
+            "console.sol"
+          )
+        );
+      }
+      const actualImportPath =
+        // @ts-ignore
+        typeof BUNDLE_VERSION !== "undefined"
+          ? path.resolve(path.join(__dirname, `console.sol`))
+          : unbundledGanacheConsoleSol;
+      const body = fse.readFileSync(actualImportPath, { encoding: "utf8" });
```

Comment:
YUNO async? Or at least cache it? Otherwise this file read will happen for not just every _file_ that imports `console.sol`, but every separate compilation of ever file that imports `console.sol` (like if a helper util uses `console.sol` which is then used by multiple contracts).

Optionally, you could inline `console.sol` when webpacking by using the define plugin to replace this `body` with the contents at bundle time. But that's weird.

Reply:
I think caching is the best approach, and would prefer to cache at the resolver interface, this function's caller. Trade memory for disk-access for all the sources.

Reply:
Ok. Make an issue for this and we can consider this comment resolved!

Reply:
#5757, if you please.